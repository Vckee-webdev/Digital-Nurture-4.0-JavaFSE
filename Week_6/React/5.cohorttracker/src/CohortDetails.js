import React from "react";
import styles from "./cohortdetais.module.css"; 

function CohortDetails({ name, StartedOn, CurrentStatus, Coach, Trainer }) {
  const statusColor = CurrentStatus === "Ongoing" ? "green" : "blue"; 

  return (
    <div className={styles.box}> 
      <h3 style={{ color: statusColor }}>{name}</h3> 
      <dl>
        <dt>StartedOn</dt>
        <dd>{StartedOn}</dd>
        <dt>CurrentStatus</dt>
        <dd>{CurrentStatus}</dd>
        <dt>Coach</dt>
        <dd>{Coach}</dd>
        <dt>Trainer</dt>
        <dd>{Trainer}</dd>
      </dl>
    </div>
  );
}

export default CohortDetails;
