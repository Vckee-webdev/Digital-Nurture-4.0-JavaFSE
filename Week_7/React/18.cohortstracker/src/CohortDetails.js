function CohortDetails({ cohort = {} }) {
  return (
    <div>
      <h3>
        {cohort.cohortCode} -
        <span>{cohort.technology}</span>
      </h3>
      <dl>
        <dt>Start Date:</dt>
        <dd>{cohort.startDate}</dd>
        <dt>End Date:</dt>
        <dd>{cohort.endDate}</dd>
      </dl>
    </div>
  );
}

export default CohortDetails;
