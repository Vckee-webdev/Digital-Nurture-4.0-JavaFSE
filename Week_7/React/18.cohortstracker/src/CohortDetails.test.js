// CohortDetails.test.js
import React from 'react';
import { shallow, mount } from 'enzyme';
import CohortDetails from './CohortDetails';

const mockCohort = {
  cohortCode: 'A2025',
  technology: 'ReactJS',
  startDate: '2025-01-01',
  endDate: '2025-06-01',
};

describe('Cohort Details Component', () => {
  test('should create the component', () => {
    const wrapper = shallow(<CohortDetails cohort={mockCohort} />);
    expect(wrapper.exists()).toBe(true);
  });

  test('should initialize the props', () => {
    const wrapper = mount(<CohortDetails cohort={mockCohort} />);
    expect(wrapper.props().cohort).toEqual(mockCohort);
  });

  test('should display cohort code in h3', () => {
    const wrapper = mount(<CohortDetails cohort={mockCohort} />);
    expect(wrapper.find('h3').text()).toContain(mockCohort.cohortCode);
  });

  test('should always render same html', () => {
    const wrapper = shallow(<CohortDetails cohort={mockCohort} />);
    expect(wrapper.html()).toMatchSnapshot();
  });
});
