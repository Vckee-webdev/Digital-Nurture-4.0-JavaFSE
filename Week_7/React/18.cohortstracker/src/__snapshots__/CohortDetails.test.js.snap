// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cohort Details Component should always render same html 1`] = `"<div><h3>A2025 -<span>ReactJS</span></h3><dl><dt>Start Date:</dt><dd>2025-01-01</dd><dt>End Date:</dt><dd>2025-06-01</dd></dl></div>"`;
